l10n_it_fatturapa_out/models/attachment.py

```diff
@@ -0,0 +1,124 @@
+        string="Out Invoices",
+        readonly=True,
+    )
+    has_pdf_invoice_print = fields.Boolean(
+        help="True if all the invoices have a printed "
+        "report attached in the XML, False otherwise.",
+        compute="_compute_has_pdf_invoice_print",
+        store=True,
+    )
+    invoice_partner_id = fields.Many2one(
+        "res.partner",
+        string="Customer",
+        store=True,
+        compute="_compute_invoice_partner_id",
+    )
+
+    _sql_constraints = [
+        (
+            "ftpa_attachment_out_name_uniq",
+            "unique(att_name)",
+            "The name of the e-invoice file must be unique!",
+        )
+    ]
+
+    @api.model
+    def get_file_vat(self):
+        company = self.env.company
+        if company.fatturapa_sender_partner:
+            if not company.fatturapa_sender_partner.vat:
+                raise UserError(
+                    _("Partner %s TIN not set.")
+                    % company.fatturapa_sender_partner.display_name
+                )
+            vat = company.fatturapa_sender_partner.vat
+        else:
+            if not company.vat:
+                raise UserError(_("Company %s TIN not set.") % company.display_name)
+            vat = company.vat
+        vat = vat.replace(" ", "").replace(".", "").replace("-", "")
+        return vat
+
+    def file_name_exists(self, file_id):
+        vat = self.get_file_vat()
+        partial_fname = r"{}\_{}.".format(vat, file_id)  # escaping _ SQL
+        # Not trying to perfect match file extension, because user could have
+        # downloaded, signed and uploaded again the file, thus having changed
+        # file extension
+        return bool(self.search([("name", "=like", "%s%%" % partial_fname)]))
+
+    @api.depends("out_invoice_ids")
+    def _compute_invoice_partner_id(self):
+        for att in self:
+            partners = att.mapped("out_invoice_ids.partner_id")
+            att.invoice_partner_id = False
+            if len(partners) == 1:
+                att.invoice_partner_id = partners.id
+
+    @api.constrains("name")
+    def _check_name(self):
+        for att in self:
+            res = self.search([("name", "=", att.name)])
+            if len(res) > 1:
+                raise UserError(_("File %s already present.") % att.name)
+
+    @api.depends("out_invoice_ids.fatturapa_doc_attachments.is_pdf_invoice_print")
+    def _compute_has_pdf_invoice_print(self):
+        """Check if all the invoices related to this attachment
+        have at least one attachment containing
+        the PDF report of the invoice"""
+        for attachment_out in self:
+            attachment_out.has_pdf_invoice_print = False
+            for invoice in attachment_out.out_invoice_ids:
+                invoice_attachments = invoice.fatturapa_doc_attachments
+                if any([ia.is_pdf_invoice_print for ia in invoice_attachments]):
+                    continue
+                else:
+                    break
```

Comment:
questo si può ridurre a:
`if **not** ... : break`

Reply:
Fatto.